### PR TITLE
DIV-4622 screen reader reads case reference ID as seperate characters

### PR DESCRIPTION
**Pull request by CoffeeOwl97**
# Description

Previously when a screen reader would read the case reference Id it would ready numeric characters as numbers, e.g. an Id of `TOM123` would be read as `TOM one hundred and twenty three`, instead of `T O M one two three`. This change adds a nunjucks filter which spaces out each of the characters which is applied to an ARIA label, so that when a screen reader tries to read it out, the version that the screen reader reads will have a whitespace inbetween each character, causing it to read out each character indevidualy.

Fixes # (issue)
[Accessibility [AA] - Case ID Number](https://tools.hmcts.net/jira/browse/DIV-4622)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Enable a screen reader and check output when on the case identification number.
Unit testing of new filter functionality.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/filters/nunjucks.test.js [hunk 1]
```diff
@@ -11,4 +11,17 @@ describe('Nunjucks insert hyphen filter', () => {
     const filteredString = nunjucksFilters.insertHyphens(caseIdNumber);
     expect(filteredString).to.equal('1509 &#8208; 0317 &#8208; 9378 &#8208; 0148');
   });
-});
\ No newline at end of file
+});
+
+describe('a11yCharSeparator', () => {
+  it('should return string formatted', () => {
```
Comment (yarpo) [inline app/filters/nunjucks.test.js]: what about empty string, number and null?

File: app/filters/nunjucks.js [hunk 2]
```diff
@@ -5,6 +5,18 @@ const newFilterInsertHyphen = input => {
   return `${input}`.match(/.{1,4}/g).join(' &#8208; ');
 };
 
-const newobjectfilter = { insertHyphens: newFilterInsertHyphen };
 
-module.exports = newobjectfilter;
\ No newline at end of file
+const a11yCharSeparator = input => {
+  if (typeof input === 'string') {
+    // Regex replace to remove all excess whitespace (e.g. '  a      b ' -> 'ab')
```
Comment (yarpo) [inline app/filters/nunjucks.js]: why not like this:

```
const removeWhitespacesRegex = /\s+/g;
const textWithoutWhitespaces = input.replace(removeWhitespacesRegex);

return returnString.split('').join(' ')
```

Isn't it clear enough now? Without comments? :)

File: app/filters/nunjucks.test.js [hunk 3]
```diff
@@ -11,4 +12,32 @@ describe('Nunjucks insert hyphen filter', () => {
...
+  });
+  it('should handle a null value', () => {
+    const filteredString = nunjucksFilters.a11yCharSeparator(null);
+    const expectedResult = null;
+    expect(filteredString).to.eql(expectedResult);
+  });
+  it('should handle a number value', () => {
+    const filteredString = nunjucksFilters.a11yCharSeparator(1234);
+    const expectedResult = 1234;
+    expect(filteredString).to.eql(expectedResult);
```
Comment (yarpo) [inline app/filters/nunjucks.test.js]: shouldn't expected result be '1234'? (string, not int)

Comment (CoffeeOwl97) [reply app/filters/nunjucks.test.js]: Test is perhaps a little deceptively named, the a11yCharSeparator is designed to seperate characters in a string. If the input is not strictly equal to a string (in this case a number), then it should just return that input as the output.

**Comment by nsingamsetty** [comment]
Tested and working fine.Merging now.

**Pull request merged**